packages/e2e-tests/cypress/integration/eligibility/householder-planning/lpd-letter-decision-date/lpd-letter-decision-date-steps.js

```diff
@@ -0,0 +1,122 @@
+import {
+  selectListedBuildingDecision
+} from '../../../../support/eligibility/listed-building/select-listed-building-decision';
+import {
+  enterDateHouseholderDecisionReceived
+} from '../../../../support/eligibility/date-decision-received/enter-date-householder-decision-received';
+const pageHeading = 'What\'s the decision date on the letter from the local planning department?';
+const pageTitle = 'What\'s the decision date on the letter from the local planning department? - Before you start - Appeal a householder planning decision - GOV.UK';
+const url = `/decision-date-householder`;
+const typeOfPlanningPageUrl = `before-you-start/type-of-planning-application`;
+const enforcementNoticePageUrl = '/enforcement-notice';
+const grantedOrRefusedPageUrl = '/granted-or-refused-householder';
+const shutterPageUrl = '/you-cannot-appeal';
+
+Given('appellant navigates to decision date received page for householder appeal',()=>{
+  goToAppealsPage(typeOfPlanningPageUrl);
+  selectPlanningApplicationType('Householder');
+  verifyPage(typeOfPlanningPageUrl);
+  clickContinueButton();
+  selectListedBuildingDecision('No');
+  clickContinueButton();
+});
+
+Given('appellant selects the {string}',(application_decision)=>{
+  selectPlanningApplicationDecision(application_decision);
+  clickContinueButton();
+});
+
+Given('appellant is on the what date was the decision received page',()=>{
+  verifyPage(url);
+  verifyPageTitle(pageTitle);
+  verifyPageHeading(pageHeading);
+});
+
+When('appellant enters the date lesser than the deadline date for {string}',(application_decision)=>{
+  if(application_decision==='Granted'){
+    const validDate = getPastDate(allowedDatePart.MONTH, 3);
+    enterDateHouseholderDecisionReceived( {day: getDate(validDate), month: getMonth(validDate) + 1, year: getYear(validDate) } );
+  }else if(application_decision==='Refused'){
+    const validDate = getPastDate(allowedDatePart.MONTH, 2);
+    enterDateHouseholderDecisionReceived( {day: getDate(validDate), month: getMonth(validDate) + 1, year: getYear(validDate) } );
+  }
+
+});
+
+When('appellant clicks on continue',()=>{
+  clickContinueButton();
+});
+
+When('appellant enters future date decision received of {string}-{string}', (datePart, value) => {
+  const futureDate = getFutureDate(datePart, value);
+  enterDateHouseholderDecisionReceived( {day: getDate(futureDate), month: getMonth(futureDate) + 1, year: getYear(futureDate) } );
+});
+
+When('appellant enters the date older than the deadline date for {string}',(application_decision)=>{
+  if(application_decision==='Granted'){
+    const pastDate = getPastDate(allowedDatePart.MONTH, 7);
+    enterDateHouseholderDecisionReceived( {day: getDate(pastDate), month: getMonth(pastDate) + 1, year: getYear(pastDate) } );
+  }else if(application_decision==='Refused'){
+    const pastDate = getPastDate(allowedDatePart.MONTH, 4);
```

Comment:
SUGGESTION:- Its appropriate to use WEEKS date part rather than MONTH date part

Reply:
Yeah he's got a point here, since not all months have the equal amount of weeks.

Reply:
updated